node/node.go

```diff
@@ -651,6 +655,21 @@ func NewNode(config *cfg.Config,
 	logger log.Logger,
 	options ...Option) (*Node, error) {
 
+	err := InitIpfs(config, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	ipfsNode, err := createIpfsNode(config, true, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create IPFS node: %w", err)
+	}
+
+	ipfsAPI, err := coreapi.NewCoreAPI(ipfsNode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create an instance of the IPFS core API: %w", err)
+	}
```

Comment:
This looks right to me. But I'm wondering if we shouldn't just pass in a `CoreAPI`object into `NewNode` instead. Then it would be easier to e.g. use other mechanisms to use IPFS. Initialization would need to happen outside of Node.

Would also like to hear what @Wondertan thinks about this.

Reply:
I like that idea better than the current implementation

Reply:
Totally agree. Having dependency initialized outside of a component is also a good practice, which for our case would allow more flexibility to allow users to use custom IPFS nodes until we decide to fully absorb IPFS(not just embed) by using its modules directly. The latter may be possible in the future as an optimization vector.

Reply:
@liamsi, @evan-forbes, but how do we then ensure that a custom IPFS node includes the needed plugin(s)? That is not clear for me now and may require some investigation. I think we can still do the proposed changed on a code level for now(pass CoreAPI to NewNode), but still not give a user ability to use a custom one, until we understand how we can ensure that, as doing that now would be a waste of time. 

@liamsi, if you haven't started this yet as part of the cleanup, I can include this change in my PR with API exposure to HTTP.

Reply:
> @liamsi, if you haven't started this yet as part of the cleanup, I can include this change in my PR with API exposure to HTTP

I have started it only in the sense that I've extracted two methods to a file for initialization and for the creation of the node. The changes that are missing is to move the creation out of node.OnStart (and similarly out of the light client constructor - but that is not a concern on master yet).
Feel free to use the changes from the light client mvp or start from scratch. It's relatively minimal changes anyways.

Regarding the plugins:

https://github.com/lazyledger/lazyledger-core/blob/d4e52d231b6db37f15b2aff2ed631461cb32fe28/p2p/ipfs.go#L153-L159

That code also exist on master (only at a different place).